command/jsonplan/values.go

```diff
@@ -171,6 +172,10 @@ func marshalPlanResources(changes *plans.Changes, ris []addrs.AbsResourceInstanc
 		ret = append(ret, resource)
 	}
 
+	sort.Slice(ret, func(i, j int) bool {
+		return ret[i].Address < ret[j].Address
```

Comment:
I know I previously said that the ordering didn't really matter here as long as its consistent, but on a second think it occurred to me that a calling program that wants to display the resources on-screen will probably want to render them in a user-friendly order but the ordering logic is pretty complicated so it'd be helpful if Terraform did the work here for programs like that.

From reading the code here it looks like that might be a little awkward to pull off since the ordering is implemented on the [`addrs.AbsResourceInstance`](https://godoc.org/github.com/hashicorp/terraform/addrs#AbsResourceInstance) type (method `Less`) and there isn't one of those in this list. So probably we should just leave this as you wrote it for now, document that the ordering is undefined but consistent in a particular version of Terraform, and then we can come back here and change the ordering to the usual user-facing one later if a need for it arises. Our immediate use-cases don't care about the order.